locust/contrib/mongo.py

```diff
@@ -0,0 +1,47 @@
+from gevent import monkey
+
+_ = monkey.patch_all()
+
+from locust import User, events
+
+import time
+
+from pymongo import MongoClient
+from pymongo.errors import PyMongoError
+
+
+class MongoDBClient:
+    def __init__(self, conn_string, db_name):
+        self.client = MongoClient(conn_string)
+        self.db = self.client[db_name]
+
+    def execute_query(self, collection_name, query):
+        start_time = time.time()
+        try:
+            response_time = int((time.time() - start_time) * 1000)
+            events.request.fire(
+                request_type="mongodb_success", name="execute_query", response_time=response_time, response_length=0
+            )
+        except PyMongoError as e:
+            response_time = int((time.time() - start_time) * 1000)
+            events.request.fire(
+                request_type="mongodb_failure",
```

Comment:
Dont change the request type on failure. just use type MONGODB and name ”QUERY”.